resteasy-core/src/main/java/org/jboss/resteasy/specimpl/UnmodifiableMultivaluedMap.java

```diff
@@ -116,6 +119,9 @@ public void clear()
    @Override
    public Set<Entry<K, List<V>>> entrySet()
    {
+      if (this.entrySet == null) {
+         this.entrySet = buildEntrySet();
+      }
```

Comment:
I have a concern with this together with the removal of final in entrySet declaration (this is not thread-safe anymore). Do you think we need to care about this and add a synchronized block here?

Reply:
I'd make the field `volatile` and stay away from synchronization. Worst case is you'll have multiple threads building the same set, but that shouldn't be a problem (they are supposed to all build the same set right?).

Reply:
I was about to comment about the result being the same even if it's created multiple times as.
I like the volatile solution, so I'll update the PR with it.

Reply:
I agree with the optimistic approach here (using ``volatile``).